### PR TITLE
90 create UI for messages jdt

### PR DESCRIPTION
**Pull request by JulienDiTria**
this creates the ui for displaying the messages between users

### CODE REVIEW

File: app/src/main/res/layout/chat_list_item.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content">
+
+    <ImageView
+        android:id="@+id/chat_user_logo"
+        android:layout_width="50dp"
+        android:layout_height="50dp"
+        android:layout_margin="@dimen/small_margin"
+        android:contentDescription="userLogo"
```
Comment (Cocograal) [inline app/src/main/res/layout/chat_list_item.xml:13]: Use string.xml file

File: app/src/main/res/layout/chat_list_item.xml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            android:id="@+id/linearLayout3"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:minHeight="25dp">
+
+            <TextView
+                android:id="@+id/chat_item_user"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:text="UserName"
```
Comment (Cocograal) [inline app/src/main/res/layout/chat_list_item.xml:32]: same here

File: app/src/main/res/layout/chat_list_item.xml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent" />
+
+            <TextView
+                android:id="@+id/chat_item_time"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:text="00:00"
```
Comment (Cocograal) [inline app/src/main/res/layout/chat_list_item.xml:41]: here as well

File: app/src/main/res/layout/chat_list_item.xml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            android:minHeight="25dp"
+            android:orientation="horizontal"
+            app:layout_constraintBottom_toTopOf="@+id/chat_item_notif"
+            app:layout_constraintTop_toTopOf="@+id/chat_item_notif">
+
+            <TextView
+                android:id="@+id/chat_item_last_msg"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:text="Last Message"
```
Comment (Cocograal) [inline app/src/main/res/layout/chat_list_item.xml:60]: same here

File: app/src/main/res/layout/chat_list_item.xml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent"
+                tools:layout_conversion_absoluteHeight="19dp"
+                tools:layout_conversion_absoluteWidth="88dp" />
+
+            <ImageView
+                android:id="@+id/chat_item_notif"
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:contentDescription="notif"
```
Comment (Cocograal) [inline app/src/main/res/layout/chat_list_item.xml:71]: same here

File: app/src/main/res/layout/fragment_chat.xml [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        app:layout_constraintBottom_toTopOf="@+id/chat_messages_recyclerView"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent">
+
+        <ImageView
+            android:id="@+id/chat_user_logo_imageview"
+            android:layout_width="100dp"
+            android:layout_height="100dp"
+            android:layout_margin="@dimen/small_margin"
+            android:contentDescription="userLogo"
```
Comment (Cocograal) [inline app/src/main/res/layout/fragment_chat.xml:24]: here as well

File: app/src/main/res/layout/fragment_chat.xml [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            tools:srcCompat="@tools:sample/avatars" />
+
+        <TextView
+            android:id="@+id/chat_username_text"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_margin="@dimen/small_margin"
+            android:text="UserName"
```
Comment (Cocograal) [inline app/src/main/res/layout/fragment_chat.xml:35]: here as well

File: app/src/main/res/layout/fragment_chat.xml [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent">
+
+        <EditText
+            android:id="@+id/chat_message_text"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_margin="@dimen/small_margin"
+            android:ems="10"
+            android:hint="Enter message"
```
Comment (Cocograal) [inline app/src/main/res/layout/fragment_chat.xml:73]: here as well

File: app/src/main/res/layout/fragment_chat.xml [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <Button
+            android:id="@+id/chat_send_button"
+            style="@style/ButtonStyle"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_margin="@dimen/small_margin"
+            android:text="Send"
```
Comment (Cocograal) [inline app/src/main/res/layout/fragment_chat.xml:87]: here as well

File: app/src/main/res/layout/message_received.xml [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content">
+
+    <TextView
+        android:id="@+id/msg_date_text"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="date"
```
Comment (Cocograal) [inline app/src/main/res/layout/message_received.xml:11]: same here

File: app/src/main/res/layout/message_received.xml [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+                android:id="@+id/msg_content_text"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:inputType="textMultiLine|text"
+                android:maxWidth="260dp"
+                android:paddingLeft="12dp"
+                android:paddingTop="8dp"
+                android:paddingRight="12dp"
+                android:paddingBottom="8dp"
+                android:text="Message Content"
```
Comment (Cocograal) [inline app/src/main/res/layout/message_received.xml:41]: here as well

File: app/src/main/res/layout/message_received.xml [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+                android:textColor="#2C358C" />
+        </LinearLayout>
+    </androidx.cardview.widget.CardView>
+
+    <TextView
+        android:id="@+id/msg_timestamp_text"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="10dp"
+        android:text="00:00"
```
Comment (Cocograal) [inline app/src/main/res/layout/message_received.xml:51]: here

File: app/src/main/res/layout/message_sent.xml [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content">
+
+    <TextView
+        android:id="@+id/msg_date_text"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="date"
```
Comment (Cocograal) [inline app/src/main/res/layout/message_sent.xml:11]: Here as well

File: app/src/main/res/layout/message_sent.xml [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+                android:id="@+id/msg_content_text"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:inputType="textMultiLine|text"
+                android:maxWidth="260dp"
+                android:paddingLeft="12dp"
+                android:paddingTop="8dp"
+                android:paddingRight="12dp"
+                android:paddingBottom="8dp"
+                android:text="Message Content" />
```
Comment (Cocograal) [inline app/src/main/res/layout/message_sent.xml:43]: here as well

**Review by Cocograal — CHANGES_REQUESTED** [review]
Overall the code is good, but make sure you put the strings in the xml file so that it will be easier to change them in the future

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/chat/ChatFragmentTest.kt [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+
+    @Test
+    fun test_user1_chat1() {
+        // Set the viewmodel's data
+        viewModel.o_currentUser = ChatsData.fakeUser1
+        viewModel.o_chat = ChatsData.fakeChat1
+
+        changeToNavChat()
+
+        Thread.sleep(1000)
```
[the remark this answers is not in the document]
Comment (JulienDiTria) [reply app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/chat/ChatFragmentTest.kt:55]: not a good idea indeed, but it s actually not needed anymore, i just forgot to remove it

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/chat/ChatFragmentTest.kt [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+
+    @Test
+    fun test_user1_emptyChat() {
+        // Set the viewmodel's data
+        viewModel.o_currentUser = ChatsData.fakeUser1
+        viewModel.o_chat = ChatsData.fakeChat1_emptyMsgs // user1 has no chats with user2
+
+        changeToNavChat()
+
+        Thread.sleep(1000)
```
[the remark this answers is not in the document]
Comment (JulienDiTria) [reply app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/chat/ChatFragmentTest.kt:122]: same

File: app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/chat/ChatListFragmentTest.kt [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+
+    private fun changeToNavChats() {
+        activityRule.onActivity { activity ->
+            activity.changeFragment(R.id.nav_chats, null)
+        }
+
+        onView(withId(R.id.chats_linearLayout)).check(matches(isDisplayed()))
+    }
+
+//    @Test
```
[the remark this answers is not in the document]
Comment (JulienDiTria) [reply app/src/androidTest/java/com/github/sdp/tarjetakuna/ui/chat/ChatListFragmentTest.kt:49]: wait for database refactoring to test this

File: app/src/main/java/com/github/sdp/tarjetakuna/model/Message.kt [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+import java.util.*
+
+class Message(
+    val id: Int,
+    val sender: User,
+    val receiver: User,
+    val content: String,
+    val timestamp: Date
+) {
+    init {
```
[the remark this answers is not in the document]
Comment (JulienDiTria) [reply app/src/main/java/com/github/sdp/tarjetakuna/model/Message.kt:12]: done

File: app/src/main/res/layout/chat_list_item.xml (hunk 2 above)
Comment (JulienDiTria) [reply app/src/main/res/layout/chat_list_item.xml:32]: done

File: app/src/main/res/layout/chat_list_item.xml (hunk 3 above)
Comment (JulienDiTria) [reply app/src/main/res/layout/chat_list_item.xml:41]: done

File: app/src/main/res/layout/fragment_chat.xml (hunk 6 above)
Comment (JulienDiTria) [reply app/src/main/res/layout/fragment_chat.xml:24]: done

File: app/src/main/res/layout/fragment_chat.xml (hunk 7 above)
Comment (JulienDiTria) [reply app/src/main/res/layout/fragment_chat.xml:35]: done

File: app/src/main/res/layout/fragment_chat.xml (hunk 8 above)
Comment (JulienDiTria) [reply app/src/main/res/layout/fragment_chat.xml:73]: done

File: app/src/main/res/layout/fragment_chat.xml (hunk 9 above)
Comment (JulienDiTria) [reply app/src/main/res/layout/fragment_chat.xml:87]: done

File: app/src/main/res/layout/message_received.xml (hunk 11 above)
Comment (JulienDiTria) [reply app/src/main/res/layout/message_received.xml:41]: done

File: app/src/main/res/layout/message_received.xml (hunk 12 above)
Comment (JulienDiTria) [reply app/src/main/res/layout/message_received.xml:51]: done

File: app/src/main/res/layout/message_sent.xml (hunk 14 above)
Comment (JulienDiTria) [reply app/src/main/res/layout/message_sent.xml:43]: done

File: app/src/main/res/layout/message_sent.xml [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+        </LinearLayout>
+    </androidx.cardview.widget.CardView>
+
+    <TextView
+        android:id="@+id/msg_timestamp_text"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="10dp"
+        android:text="00:00"
```
[the remark this answers is not in the document]
Comment (JulienDiTria) [reply app/src/main/res/layout/message_sent.xml:52]: done

**Review by Cocograal — APPROVED** [review]
Thanks for the changes!

**Pull request merged**